src/Desktop/ColorContrastAnalyzer/ImageCollection.cs

```diff
@@ -58,13 +99,15 @@ public ColorContrastResult RunColorContrastCalculation()
 
                     if (result == null) result = newResult;
 
-                    if (newResult.ConfidenceValue() == ColorContrastResult.Confidence.High)
+                    if (newResult.Confidence == Confidence.High)
                     {
                         result = newResult;
                         break;
                     }
-                    else if (newResult.ConfidenceValue() == ColorContrastResult.Confidence.Mid &&
-                      result.ConfidenceValue() == ColorContrastResult.Confidence.Low)
+                    else if (newResult.Confidence == Confidence.Mid &&
+#pragma warning disable CA1508 // Avoid dead conditional code
```

Comment:
Can you share some context around this warning? It feels like we can remove line `109` because it will always be true. By this point in the code possible values for `result` include:
- `newResult` in the current iteration (then `newResult.Confidence` must equal `result.Confidence`
- `newResult` from a previous iteration (then `result.Confidence` must be high or mid because we don't replace `result` otherwise)

Otherwise I agree it's probably good to leave the old algorithm mostly alone to preserve its semantics. If so, it might be good to explicitly mention that near the warning.

Reply:
The Analyzer claims the condition is always false, but testing in the debugger proves otherwise. I've restructured it a bit (fixed a bug in the process) and updated the comments